src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.UnifiedBuild.Tasks/SigningValidation.cs

```diff
@@ -0,0 +1,365 @@
+
+            throw new Exception($"SignCheck detected unsigned files.");
+        }
+    }
+
+    /// <summary>
+    /// Gets the command and arguments to run signcheck
+    /// </summary>
+    private (string command, string arguments) GetSignCheckCommandAndArguments()
+    {
+        string exclusionsFile = Path.Combine(DotNetRootDirectory, "eng", "SignCheckExclusionsFile.txt");
+        string sdkTaskScript = Path.Combine(DotNetRootDirectory, "eng", "common", "sdk-task");
+
+        string argumentsTemplate =
+            $"'{sdkTaskScript}.$scriptExtension$' " +
+            $"$argumentPrefix$task SigningValidation " +
+            $"$argumentPrefix$restore " +
+            $"/p:PackageBasePath='{_signCheckFilesDirectory}' " +
+            $"/p:EnableStrongNameCheck=true " +
+            $"/p:SignCheckLog='{GetLogPath(_signCheckStdoutLogFileName)}' " +
+            $"/p:SignCheckErrorLog='{GetLogPath(_signCheckStderrLogFileName)}' " +
+            $"/p:SignCheckResultsXmlFile='{GetLogPath(_signCheckResultsXmlFileName)}' " +
+            $"/p:SignCheckExclusionsFile='{GetSignCheckExclusionsFile()}' " +
+            $"$additionalArgs$";
+
+        string command = string.Empty;
+        string arguments = string.Empty;
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            command = "powershell.exe";
+            string formattedArguments = argumentsTemplate
+                .Replace("$scriptExtension$", "ps1")
+                .Replace("$argumentPrefix$", "-")
+                .Replace("$additionalArgs$", "-msbuildEngine vs");
+            arguments = $"& \"{formattedArguments}\"";
+        }
+        else
+        {
+            command = "/bin/bash";
+            string formattedArguments = argumentsTemplate
+                .Replace("$scriptExtension$", "sh")
+                .Replace("$argumentPrefix$", "--")
+                .Replace("$additionalArgs$", string.Empty);
+            arguments = $"-c \"{formattedArguments}\"";
+        }
+
+        return (command, arguments);
+    }
+
+    /// <summary>
+    /// Extracts the value of the specified attribute from the element and logs an error if it's missing or empty.
+    /// </summary>
+    private string ExtractAttribute(XElement element, string attributeName)
+    {
+        string? value = element.Attribute(attributeName)?.Value;
+        if (string.IsNullOrEmpty(value))
+        {
+            Log.LogError($"{attributeName} is null or empty in element: {element}");
+        }
+        return value ?? string.Empty;
+    }
+
+    /// <summary>
+    /// Gets the path to the log file in the output logs directory.
+    /// </summary>
+    private string GetLogPath(string fileName)
+        => Path.Combine(OutputLogsDirectory, fileName);
+
+    /// <summary>
+    /// Gets the SignCheckExclusionsFile path.
+    /// Adds exclusions to the file and writes it to a temporary location.
+    /// </summary>
+    private string GetSignCheckExclusionsFile()
```

Comment:
When we move this functionality into arcade, we could allow specification of multiple exclusion files. Then the YAML would conditionally pass the DAC signing exclusion file based on build branch.